### DISCUSSION TITLE
1.1.0

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#110] [Обновления]
### **Расширение необходимо вручную переустановить из папки `dist` на главной странице репозитория**

Решился, всё-таки, попробовать перенести расширение на Webpack, дабы не тратить кучу время на внесение одних и тех же изменений в двух разных версиях расширения. Вроде бы, перенес успешно, и, теперь, должно быть намного удобнее и быстрее разрабатывать расширение. Единственный минус, у тех, кто раньше ставил расширение отвалятся автообновления от Tampermonkey (или другого расширения для установки юзерскриптов) из-за смены путей к файлу. 

Минимальные тесты провел - всё работает, но в любом случае, сообщайте о всех найденных ошибках после обновления. Так же, в описание добавил инструкцию по сборке - для тех, кто захочет что-то изменить в коде расширения.

- Временно убраны английские названия плагинов из-за не совместимости с плагином WebpackUserscript (3.1.0)
- Теперь, vot-cloudflare.user.js и vot.user.js собираются из одного единого файла с помощью webpack, что убирает необходимость повторно писать один и тот же код. Готовые расширения перемещены в папку "dist" (расширения не сжаты)
- Убрано немного мусора из кода
- Добавлена поддержка Twitter

## What's Changed
* Update youtube algorithm. Added Rehike support by @ten4dinosaur in https://github.com/ilyhalight/voice-over-translation/pull/100

## New Contributors
* @ten4dinosaur made their first contribution in https://github.com/ilyhalight/voice-over-translation/pull/100

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.0.9.9...1.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.1.0'>1.1.0</a>.</em>

**Comment by FiLinX** [comment]
ничего не понял про пертурбаций произведённых)  и пока подержу 1.0.9.9 отключенным в запасе..
1.1.0 установил, подправил defaultVideoVolume на = 0.05, проверил пока лишь ютуб - работает! )
хроме87

**Reply by ilyhalight** [reply]
> ничего не понял про пертурбаций произведённых) и пока подержу 1.0.9.9 отключенным в запасе.. 1.1.0 установил, подправил defaultVideoVolume на = 0.05, проверил пока лишь ютуб - работает! ) хроме87

в целом-то ничего толком и не поменялось в исходниках расширения, единственное, что будет заметно обычному пользователю, то, что код готового расширения, теперь, как-будто забит каким-то непонятным мусором из-за сборки через вебпак

**Comment by CrescentXIII** [comment]
Спасибо, Вам, за вашу работу! 
Подскажите, пожалуйста, а в чем разница между VOT и VOT cloudfдare?  На опере проверил, вроде обе одинаково работают =) 
И уточнить хотел, для работы , как я заметил на практике, нужен тольк один файл из всего архива vot.user.js?
Спасибо еще раз!

**Reply by ilyhalight** [reply]
> Спасибо, Вам, за вашу работу! Подскажите, пожалуйста, а в чем разница между VOT и VOT cloudfдare? На опере проверил, вроде обе одинаково работают =) И уточнить хотел, для работы , как я заметил на практике, нужен тольк один файл из всего архива vot.user.js? Спасибо еще раз!

С некоторыми расширениями/браузерами обычная версия VOT не работает - запрос не проходит. Поэтому в этом случае необходимо использовать cloudflare версию.

Для установки нужен только vot.user.js или vot-cloudflare.user.js

**Comment by alladyn40** [comment]
Меню перевода появляется даже в русскоязычных роликах.
Можно это как то исправить, иначе при включении функции автоматического перевода пытается перевести русскоязычный ролик заглушая звук?
![image](https://user-images.githubusercontent.com/130770281/232067519-108dbf12-678c-44c8-a08c-00b00a780ca2.png)

**Reply by ilyhalight** [reply]
Немного позже добавлю такую опцию для ютуба

**Reply by FiLinX** [reply]
[alladyn40](https://github.com/alladyn40)  в офтопе, но поделись плиз сцылкой на этого Сидоровича, откуда это? )

[ilyhalight](https://github.com/ilyhalight)  а разве у ютуба ролики имеют какие то ланг-пометки?

**Reply by ilyhalight** [reply]
> [ilyhalight](https://github.com/ilyhalight) а разве у ютуба ролики имеют какие то ланг-пометки?

Можно через данные видеоплеера получать заголовок, описание и ник автора ролика, и уже потом их проверять на наличие русских символов. Это может быть довольно не точно, поэтому я и хочу это сделать как дополнительную опцию, а не делать ее принудительной.
![изображение](https://user-images.githubusercontent.com/62353659/232306932-de02b6e8-f6e8-4011-8ab7-016571c9dc1b.png)
![изображение](https://user-images.githubusercontent.com/62353659/232307103-e4b03b1d-11aa-47e4-a42d-70f538855eaf.png)

**Reply by ilyhalight** [reply]
> > [ilyhalight](https://github.com/ilyhalight) а разве у ютуба ролики имеют какие то ланг-пометки?
> 
> Можно через данные видеоплеера получать заголовок, описание и ник автора ролика, и уже потом их проверять на наличие русских символов. Это может быть довольно не точно, поэтому я и хочу это сделать как дополнительную опцию, а не делать ее принудительной. ![изображение](https://user-images.githubusercontent.com/62353659/232306932-de02b6e8-f6e8-4011-8ab7-016571c9dc1b.png)

А языковых пометок каких-либо к сожалению не нашёл

UPD: Чуть-чуть еще полазил и нашел возможность получать язык видео через субтитры как более точную альтернативу (правда они есть не на всех роликах)

**Reply by alladyn40** [reply]
> [alladyn40](https://github.com/alladyn40) в офтопе, но поделись плиз сцылкой на этого Сидоровича, откуда это? )

[BadComedian] - Чернобыль (РОССИЙСКИЙ ОТВЕТ HBO)
